### PR TITLE
[bugfix] Do not ignore CORS violations

### PR DESCRIPTION
**Pull request by pboos**
Remove CORS trickery that was added as part of [this bug ticket](https://github.com/microsoft/playwright/issues/2641).
In my opinion having such trickery is bad practice and we already had 2 incidents that should would have been detected if this trickery wasn't there (luckily Testcafe still detected them as we have not migrated all tests to playwright yet). In both incidents a header was added accidentally which caused CORS violations due to the backend not having these allowed in the CORS headers. This should be detected in e2e tests. Also in Testcafe CORS errors are not ignored if one adds request hooks.

Reasons in my opinion why this CORS trickery is bad:
- This hides real problems (as we learned the hard way)
- It is inconsistent over browsers, currently only chrome has this trickery. In Firefox (and maybe others) the CORS errors appear.
- If a CORS error appears in the test (as in the bug ticket) then this is a VALID error! It should not be disabled. If one wants to get rid of it there are other ways, but this should not be enforced for everyone using the e2e framework. The way those CORS errors can be ignored if they really want to:
    - They setup a proxy that routes the requests and there disables CORS
    - If the problem is that they run the tests against localhost instead of production -> run local server with self signed ssl certificate, add that certificate to be accepted on the host that runs the tests, adjust hosts to point the production url to 127.0.0.1

See issue: https://github.com/microsoft/playwright/issues/27339

### CODE REVIEW

**Comment by pboos** [comment]
@microsoft-github-policy-service agree [company="GetYourGuide AG"]

**Comment by pboos** [comment]
@microsoft-github-policy-service agree company="GetYourGuide AG"

**Comment by pboos** [comment]
```
@microsoft-github-policy-service agree company="GetYourGuide AG"
```

**Comment by pboos** [comment]
@microsoft-github-policy-service agree company="getyourguide"

File: packages/playwright-core/src/server/chromium/crNetworkManager.ts [hunk 1]
```diff
@@ -255,30 +255,6 @@ export class CRNetworkManager {
       frame = this._page._frameManager.frameAttached(requestWillBeSentEvent.frameId, null);
     }
 
-    // CORS options preflight request is generated by the network stack. If interception is enabled,
-    // we accept all CORS options, assuming that this was intended when setting route.
-    //
-    // Note: it would be better to match the URL against interception patterns.
```
Comment (yury-s) [inline packages/playwright-core/src/server/chromium/crNetworkManager.ts:261]: Let's implement this url matching instead. That way you would fulfill all the analytics requests that you mentioned without going to the server while keeping the rest unaffected by the interception. And then all the tests (and the use cases they cover) that broke will keep working.

**Review by yury-s — CHANGES_REQUESTED** [review]

**Comment by yury-s** [comment]
Closing as part of PR cleanup. We are not ready to accept it in the current form as it will break too many existing clients. Let's continue discussion in the bug.

**Pull request closed without merging**